### PR TITLE
sbcl: bootstrap from CLISP on supported systems

### PR DESCRIPTION
**Pull request by hraban**
## Description of changes

Don't use the binary blob in the build phase, where possible.

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/compilers/sbcl/bootstrap.nix [hunk 1]
```diff
@@ -2,31 +2,13 @@
...
       system = "x86-linux";
       sha256 = "07f3bz4br280qvn85i088vpzj9wcz8wmwrf665ypqx181pz2ai3j";
     };
-    aarch64-linux = {
-      version = "1.3.16";
-      system = "arm64-linux";
-      sha256 = "0q1brz9c49xgdljzfx8rpxxnlwhadxkcy5kg0mcd9wnxygind1cl";
-    };
     armv7l-linux = {
       version = "1.2.14";
```
Comment (hraban) [inline pkgs/development/compilers/sbcl/bootstrap.nix:13]: I've been conservative in what I remove here but honestly I'd be curious to see if any of these odd platforms still compile at all. Is anybody even testing them?

**Comment by hraban** [comment]
@ofborg sbcl

**Comment by hraban** [comment]
@ofborg build sbcl

**Comment by hraban** [comment]
Why does it say "Skipped — No attempt"?

**Review by 7c6f434c — APPROVED** [review]

File: pkgs/development/compilers/sbcl/bootstrap.nix (hunk 1 above)
Comment (7c6f434c) [reply pkgs/development/compilers/sbcl/bootstrap.nix]: At least for FreeBSD, Nixpkgs support for the platform previously existed but is currently broken.  There are some attempts to restore this support and get it right this time. I expect CLISP to work there fine once we have general platform support.

**Comment by 7c6f434c** [comment]
Hm, segfault at x86_64-darwin… should we use boostrap SBCL there then until someone with a machine investigates?

(aarch64-linux reminds us that the change slows down the bootstrap, but I guess we can tolerate the regression…)

**Comment by hraban** [comment]
Can we bump the timeout somehow? Doesn't this build failure mean it's not in the binary cache, now?

**Comment by hraban** [comment]
@ofborg build sbcl

**Comment by 7c6f434c** [comment]
Ofborg timeout for CI, r-ryantm timeout for update rev-deps, and Hydra timeout for filling binary cache are completely independent

**Comment by hraban** [comment]
Thanks that makes sense in retrospect but I never realized that

**Comment by hraban** [comment]
I built this at home on a x86_64-linux (NixOS) VM in UTM and it built fine so I'm confused as to why it's segfaulting 🤷‍♀️

**Comment by 7c6f434c** [comment]
Well, irreproducible segfaults wouldn't be that surprising…

@ofborg build sbcl

**Comment by hraban** [comment]
Never mind I can't read. x86_64-darwin segfaults here too. I'm trying ECL instead see if that works...

**Pull request merged**